toolchain/check/handle_function.cpp

```diff
@@ -278,6 +278,11 @@ static auto BuildFunctionDecl(Context& context,
     self_param_id = *i;
   }
 
+  if (virtual_modifier != SemIR::Function::VirtualModifier::None &&
+      !self_param_id.has_value()) {
+    CARBON_DIAGNOSTIC(VirtualWithoutSelf, Error, "virtual non-instance method");
```

Comment:
```suggestion
    CARBON_DIAGNOSTIC(VirtualWithoutSelf, Error, "virtual class function");
```

Terminology comes from:
https://github.com/carbon-language/carbon-lang/blob/trunk/docs/design/classes.md?plain=1#L850

> Member functions can either be class functions or methods. Class functions are
> members of the type, while methods can only be called on instances.

(on my mind because we were recently discussing this terminology)